pi/models/unet.py

```diff
@@ -0,0 +1,11078 @@
+import importlib
+import json
+import logging
+import sys
+from pathlib import Path, PosixPath
+
+logger = logging.getLogger(__name__)
+
+import functools
+import operator as op
+import os
+from dataclasses import dataclass, fields
+from collections import OrderedDict, defaultdict
+from copy import deepcopy
+from functools import partial
+from torch import Tensor, device
+from typing import Any, Callable, Dict, List, Optional, Tuple, Union
+from uuid import uuid4
+import inspect
+import math
+import numpy as np
+import re
+import torch.nn as nn
```

Comment:
Is this still taking a torch dependency or is there some magic that I'm not understanding (just out of curiosity)? Or by "no external dependencies" do you mean self contained to torch.nn and then the idea is to plugin whatever replacement you want.

Reply:
this isn't done yet - this first WIP pass import for `torch.nn`; then after cleanup it'll import from `pi.nn`.

Reply:
I see, makes sense